### PR TITLE
Add new features on app

### PR DESCRIPTION
**Pull request by bh-2024**
Here are changes added on react component and also python change

### CODE REVIEW

File: Indie_Gems_Portal/scripts/sort_genres.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# File: process_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres.py:1]: Unfortunately, I don't have any code provided in the prompt. However, if you provide the code from `Indie_Gems_Portal/scripts/sort_genres.py`, I'd be happy to help summarize the code review suggestions and highlight the most important considerations.

Please paste the code, and I'll assist you with a thorough summary of the review findings!

File: Indie_Gems_Portal/scripts/sort_genres_test [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# File: test_sort_genres.py
```
Comment (bh-2024) [inline Indie_Gems_Portal/scripts/sort_genres_test:1]: The most important considerations for the code review are:

1. **Lack of comprehensive edge case testing**: The current tests only cover basic scenarios but do not thoroughly test all possible edge cases, such as invalid input data or duplicate genres.
2. **Use of hardcoded mock file path**: Using a hardcoded mock file path can lead to confusion or errors if the file path needs to be updated in the future.
3. **Unused `setUp` method**: The unused `setUp` method should be removed to avoid confusion.

To address these concerns, it is recommended to:

1. Add more comprehensive tests for invalid input data and edge cases.
2. Use a flexible mock file path configuration, such as making it configurable or dynamically generating it within the test.
3. Remove the unused `setUp` method.

Overall, the code review suggests that while the code structure is consistent with project naming conventions, there are several areas that need improvement to ensure comprehensive testing and flexibility in the code.

File: Indie_Gems_Portal/src/components/GenreList.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import {
   Spinner,
   Text,
 } from "@chakra-ui/react";
+import React, {useEffect, useState} from "react";
```
Comment (bh-2024) [inline Indie_Gems_Portal/src/components/GenreList.tsx:11]: Here are the most important considerations from the code review:

1. **Consistent naming conventions**: Use a consistent naming convention throughout the component.
2. **Extract utility functions**: Consider extracting utility functions, such as data transformation or filtering, into separate files or modules to improve modularity and reusability.
3. **Robust error handling**: Implement more robust error handling mechanisms, including displaying an error message or retrying the fetch request.
4. **State management**: Consider using a more robust state management approach (e.g., Redux, MobX) for larger applications.
5. **Caching and performance optimization**: Implement caching mechanisms to reduce repeated fetches and optimize DOM manipulation by reusing existing elements.
6. **Accessibility and usability**: Ensure that the component provides adequate ARIA attributes and follows accessibility guidelines (WCAG 2.1).

Additionally, the reviewer recommends:

1. Reviewing and refactoring the code according to best practices.
2. Considering state management libraries like Redux or MobX for larger applications.

A sample refactored example is provided, showcasing improvements in data fetching logic, error handling, and input validation.
